subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -807,32 +807,43 @@ bool ull_conn_peer_connected(uint8_t own_addr_type, uint8_t *own_addr,
 void ull_conn_setup(memq_link_t *link, struct node_rx_hdr *rx)
 {
 	struct node_rx_ftr *ftr;
+	struct ll_conn *conn;
 	struct lll_conn *lll;
 
 	ftr = &(rx->rx_ftr);
+	lll = CONN_HDR2LLL(ftr->param);
 
-	/* NOTE: LLL conn context SHALL be after lll_hdr in
-	 *       struct lll_adv and struct lll_scan.
-	 */
-	lll = *((struct lll_conn **)((uint8_t *)ftr->param +
-				     sizeof(struct lll_hdr)));
-	switch (lll->role) {
+	if (lll) {
+		conn = lll->hdr.parent;
+		if ((lll == &conn->lll) && (lll->handle == 0xffff)) {
```

Comment:
This is a bit "over the top" and not necessary for this particualr race, but always nice to be safe...